### PR TITLE
[Fix] Fix several errors in docstring

### PR DESCRIPTION
**Pull request by THU17cyz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
I find several bugs in the auto-generated api docs in [readthedocs](https://mmdetection3d.readthedocs.io/en/latest/).

## Modification
I will describe the modifications in several comments below.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/core/bbox/iou_calculators/iou3d_calculator.py [hunk 1]
```diff
@@ -176,17 +176,17 @@ def __call__(self, bboxes1, bboxes2, mode='iou', is_aligned=False):
 
         Args:
             bboxes1 (Tensor): shape (B, m, 6) in <x1, y1, z1, x2, y2, z2>
-            format or empty.
+                format or empty.
```
Comment (THU17cyz) [inline mmdet3d/core/bbox/iou_calculators/iou3d_calculator.py:179]: **Indentation**
Before fix:
image

After fix:
image

File: mmdet3d/core/bbox/iou_calculators/iou3d_calculator.py [hunk 2]
```diff
@@ -176,17 +176,17 @@ def __call__(self, bboxes1, bboxes2, mode='iou', is_aligned=False):
...
             bboxes2 (Tensor): shape (B, n, 6) in <x1, y1, z1, x2, y2, z2>
-            format or empty.
+                format or empty.
                 B indicates the batch dim, in shape (B1, B2, ..., Bn).
-                If ``is_aligned `` is ``True``, then m and n must be equal.
```
Comment (THU17cyz) [inline mmdet3d/core/bbox/iou_calculators/iou3d_calculator.py:183]: **Space before the second "``"**
Before fix:
image

After fix:

File: mmdet3d/core/bbox/structures/base_box3d.py [hunk 3]
```diff
@@ -199,7 +199,7 @@ def convert_to(self, dst, rt_mat=None):
         """Convert self to ``dst`` mode.
 
         Args:
-            dst (:obj:`BoxMode`): The target Box mode.
```
Comment (THU17cyz) [inline mmdet3d/core/bbox/structures/base_box3d.py:202]: **Wrong object name**
Before fix:
image

After fix:
image
The `Box3DMode` after fix is clickable, which links to the definition of this class

File: mmdet3d/core/bbox/structures/base_box3d.py [hunk 4]
```diff
@@ -293,10 +293,10 @@ def cat(cls, boxes_list):
         """Concatenate a list of Boxes into a single Boxes.
 
         Args:
-            boxes_list (list[:obj:`BaseInstances3DBoxes`]): List of boxes.
+            boxes_list (list[:obj:`BaseInstance3DBoxes`]): List of boxes.
 
         Returns:
-            :obj:`BaseInstances3DBoxes`: The concatenated Boxes.
```
Comment (THU17cyz) [inline mmdet3d/core/bbox/structures/base_box3d.py:299]: Typos

File: mmdet3d/datasets/lyft_dataset.py [hunk 5]
```diff
@@ -49,7 +49,7 @@ class LyftDataset(Custom3DDataset):
             Defaults to True.
         test_mode (bool, optional): Whether the dataset is in test mode.
             Defaults to False.
-    """
+    """  # noqa: E501
```
Comment (THU17cyz) [inline mmdet3d/datasets/lyft_dataset.py:52]: **Removing noqa from api docs**
Before fix:
image

After fix:
image

Reference: https://github.com/pyjanitor-devs/pyjanitor/issues/555

File: mmdet3d/models/detectors/fcos_mono3d.py [hunk 6]
```diff
@@ -4,11 +4,11 @@
 
 @DETECTORS.register_module()
 class FCOSMono3D(SingleStageMono3DDetector):
-    r"""FCOS3D <https://arxiv.org/abs/2104.10956>`_ for monocular 3D object detection.
+    r"""`FCOS3D <https://arxiv.org/abs/2104.10956>`_ for monocular 3D object detection.
```
Comment (THU17cyz) [inline mmdet3d/models/detectors/fcos_mono3d.py:7]: Typo

**Review by Wuziyi616 — APPROVED** [review]

File: mmdet3d/datasets/lyft_dataset.py (hunk 5 above)
Comment (Wuziyi616) [reply mmdet3d/datasets/lyft_dataset.py:52]: 👍

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**